File: scripts/check_riya_sonu.py
import sqlite3
import os
import glob
import time
from datetime import datetime

db_paths = [
    "/vps/ess-os/prisma/dev.db",
    "/vps/ess-os/dev.db",
    "/vps/ess-os/prisma/prod.db",
    "/vps/ess-os/prod.db",
    "./prisma/dev.db",
    "./dev.db"
]

db_file = None
for p in db_paths:
    if os.path.exists(p):
        db_file = p
        break

if not db_file:
    found = glob.glob("**/*.db", recursive=True)
    if found:
        db_file = found[0]

print(f"Using database: {db_file}")
if not db_file:
    print("Database not found!")
    exit(1)

conn = sqlite3.connect(db_file)
cur = conn.cursor()

print("--- USERS ---")
cur.execute("SELECT u.id, u.name, u.email, e.id, e.employeeIdCode FROM User u LEFT JOIN Employee e ON e.userId = u.id")
users = cur.fetchall()
for u in users:
    print(u)

print("\n--- ATTENDANCES FOR LAST 7 DAYS ---")
cur.execute("""
    SELECT a.id, u.name, e.employeeIdCode, a.date, a.punchIn, a.punchOut, a.status, a.totalMinutes, a.punchOutReason, a.punchOutRequestStatus
    FROM Attendance a
    JOIN Employee e ON a.employeeId = e.id
    JOIN User u ON e.userId = u.id
    ORDER BY a.date DESC
    LIMIT 30
""")
for r in cur.fetchall():
    print(r)

conn.close()

File: scripts/check_riya_sonu.ts
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

async function main() {
  console.log("=== EMPLOYEES & USERS ===");
  const users = await prisma.user.findMany({
    include: {
      employeeProfile: true,
    },
  });
  for (const u of users) {
    console.log(`User: ${u.id} | ${u.name} | ${u.email} | EmpID: ${u.employeeProfile?.id} (${u.employeeProfile?.employeeIdCode})`);
  }

  console.log("\n=== RECENT ATTENDANCE RECORDS (Past 30) ===");
  const attendances = await prisma.attendance.findMany({
    take: 30,
    orderBy: { date: "desc" },
    include: {
      employee: {
        include: { user: true },
      },
    },
  });

  for (const a of attendances) {
    console.log(
      `ID: ${a.id} | Name: ${a.employee?.user?.name} (${a.employee?.employeeIdCode}) | Date: ${a.date.toISOString()} | PunchIn: ${a.punchIn?.toISOString()} | PunchOut: ${a.punchOut?.toISOString()} | TotalMin: ${a.totalMinutes} | Status: ${a.status} | ReqStatus: ${a.punchOutRequestStatus}`
    );
  }
}

main()
  .catch((e) => {
    console.error(e);
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
